hicommonwealth/commonwealth
# Move server controller methods into separate files

## Link to Issue
Closes: #4459

## Description of Changes
- No changes to business logic.
- Moves the server controller method implementations into individual files, to make them easier to work with.
- Using a consistent pattern where each method has an Options type for arguments and Result type for return value
- Classes are still used to hold contextual objects like DB model, ban cache, token balance cache, etc.

## Test Plan
- All unit tests pass for the affected functionality, so everything *should* work without regressions
- We should still manually CA test these affected features (I will leave a comment when I've finished doing this):
  - Create comment reaction
  - Delete comment
  - View comment reactions
  - Search comments
  - Update comment
  - Delete reaction
  - Create thread comment
  - Create thread reaction
  - Create thread
  - Delete thread
  - View active threads (dashboard home page)
  - View bulk threads (dashboard home page)
  - Search threads
  - Update thread


## Deployment Plan
N/A

## Other Considerations
N/A
